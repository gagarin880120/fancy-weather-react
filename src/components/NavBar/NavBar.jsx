import React from 'react';
import styles from './NavBar.module.css';
import TemperatureScaleButton from '../TemperatureScaleButton/TemperatureScaleButton';
import LanguageSelectContainer from '../LanguageSelect';
import SearchField from '../SearchField/SearchField';

export default function NavBar() {
  return (
    <div className={styles.wrapper}>
      <div className={styles.container}>
        <LanguageSelectContainer />
        <div className={styles.scaleButtons}>
          <TemperatureScaleButton
            scale="celsius"
          />
          <TemperatureScaleButton
            scale="fahrenheit"
          />
        </div>
      </div>
      <SearchField />
    </div>
  );
}
