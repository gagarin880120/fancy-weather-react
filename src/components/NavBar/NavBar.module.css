.wrapper {
  padding: 10px 0;
  display: flex;
  justify-content: space-between;
}

.container {
  width: 8%;
  display: flex;
  justify-content: space-between;
}
