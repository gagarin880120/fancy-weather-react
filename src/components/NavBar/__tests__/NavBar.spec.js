import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import NavBar from '../NavBar';

describe('NavBar', () => {
  describe('component', () => {
    test('should render without crashing', () => {
      const renderer = new ShallowRenderer();
      renderer.render(<NavBar />);

      const result = renderer.getRenderOutput();

      expect(result).toMatchSnapshot();
    });
  });
});
