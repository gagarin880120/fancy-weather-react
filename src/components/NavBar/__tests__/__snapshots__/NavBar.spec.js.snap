// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavBar component should render without crashing 1`] = `
<div
  className="wrapper"
>
  <div
    className="container"
  >
    <Memo(Connect(LanguageSelectContainer)) />
    <div
      className="scaleButtons"
    >
      <Memo(Connect(TemperatureScaleButtonContainer))
        scale="celsius"
      />
      <Memo(Connect(TemperatureScaleButtonContainer))
        scale="fahrenheit"
      />
    </div>
  </div>
  <Memo(Connect(SearchFieldContainer)) />
</div>
`;
